js/serializers/questions.js

```diff
@@ -85,6 +88,9 @@ define([
           var booleanParameters = [
             hasUserAnswer,
             isUserAnswerArray,
+            hasAttemptStates,
+            isAttemptStatesArray,
+            component['_isComplete'],
```

Comment:
It would be nice - but obviously by no means essential - to switch any instances of these that don't need to be written like this over to dot-notation.
```suggestion
            component._isComplete,
```

Reply:
I half-offered to @oliverfoster that I'd refactor `questions.js` at same point; it definitely needs a bit of TLC.